admin/resources/js/components/Table.tsx

```diff
@@ -41,27 +106,101 @@ function Table<T extends Record<string, unknown>>({
         ) : null}
         {headerGroups.map((headerGroup) => (
           <tr {...headerGroup.getHeaderGroupProps()}>
-            {headerGroup.headers.map((column) => (
-              <th
-                {...column.getHeaderProps(column.getSortByToggleProps())}
-                key={column.id}
+            {headerGroup.headers.map(
+              (column) =>
+                shouldBeVisible(column.Header) && (
+                  <th
+                    {...column.getHeaderProps(column.getSortByToggleProps())}
+                    key={column.id}
+                  >
+                    {column.render("Header")}
+                    <span>
+                      {column.isSorted && (column.isSortedDesc ? " ▼" : " ▲")}
+                    </span>
+                  </th>
+                ),
+            )}
+            <th>
+              <button
```

Comment:
I asked Jerbo where this button should be placed. We decided: to the right of the GlobalFilter search button, for now.

Reply:
And like the GlobalFilter component, this seems like a good chunk of code to extract to its own component. Could be called VisibleColumnSelector, or something like that.